pkg/provider/azure/keyvault/keyvault.go

```diff
@@ -201,13 +215,270 @@ func (a *Azure) ValidateStore(store esv1beta1.GenericStore) error {
+		}
+	}
+	return nil
+}
+
+func (a *Azure) deleteKeyVaultCertificate(ctx context.Context, certName string) error {
+	value, err := a.baseClient.GetCertificate(ctx, *a.provider.VaultURL, certName, "")
+	ok, err := canDelete(value.Tags, err)
+	if err != nil {
+		return fmt.Errorf("error getting certificate %v: %w", certName, err)
+	}
+	if ok {
+		_, err = a.baseClient.DeleteCertificate(ctx, *a.provider.VaultURL, certName)
+		if err != nil {
+			return fmt.Errorf("error deleting certificate %v: %w", certName, err)
+		}
+	}
+	return nil
+}
+
 func (a *Azure) DeleteSecret(ctx context.Context, remoteRef esv1beta1.PushRemoteRef) error {
-	return fmt.Errorf("not implemented")
+	objectType, secretName := getObjType(esv1beta1.ExternalSecretDataRemoteRef{Key: remoteRef.GetRemoteKey()})
+	switch objectType {
+	case defaultObjType:
+		return a.deleteKeyVaultSecret(ctx, secretName)
+	case objectTypeCert:
+		return a.deleteKeyVaultCertificate(ctx, secretName)
+	case objectTypeKey:
+		return a.deleteKeyVaultKey(ctx, secretName)
+	default:
+		return fmt.Errorf("secret type '%v' is not supported", objectType)
+	}
 }
 
-// Not Implemented PushSecret.
+func getCertificateFromValue(value []byte) (*x509.Certificate, error) {
+	_, localCert, err := pkcs12.Decode(value, "")
+	if err != nil {
+		pemBlock, _ := pem.Decode(value)
+		if pemBlock == nil {
+			return x509.ParseCertificate(value)
+		}
+		return x509.ParseCertificate(pemBlock.Bytes)
+	}
+	return localCert, err
+}
+
+func getKeyFromValue(value []byte) (interface{}, error) {
+	val := value
+	pemBlock, _ := pem.Decode(value)
+	// if a private key regular expression doesn't match, we should consider this key to be symmetric
+	if pemBlock == nil {
+		return val, nil
+	}
+	val = pemBlock.Bytes
+	switch pemBlock.Type {
+	case "PRIVATE KEY":
+		return x509.ParsePKCS8PrivateKey(val)
+	case "RSA PRIVATE KEY":
+		return x509.ParsePKCS1PrivateKey(val)
+	case "EC PRIVATE KEY":
+		return x509.ParseECPrivateKey(val)
+	default:
+		return nil, fmt.Errorf("key type %v is not supported", pemBlock.Type)
+	}
+}
+
+func canCreate(tags map[string]*string, err error) (bool, error) {
+	aerr := &autorest.DetailedError{}
+	conv := errors.As(err, aerr)
+	if err != nil && !conv {
+		return false, fmt.Errorf("could not parse error: %w", err)
+	}
+	if conv && aerr.StatusCode != 404 {
+		return false, fmt.Errorf("unexpected api error: %w", err)
+	}
+	if err == nil {
+		manager, ok := tags["managed-by"]
+		if !ok || manager == nil || *manager != managerLabel {
+			return false, fmt.Errorf("not managed by external-secrets")
+		}
+	}
+	return true, nil
+}
+
+func (a *Azure) setKeyVaultSecret(ctx context.Context, secretName string, value []byte) error {
+	secret, err := a.baseClient.GetSecret(ctx, *a.provider.VaultURL, secretName, "")
+	ok, err := canCreate(secret.Tags, err)
+	if err != nil {
+		return fmt.Errorf("cannot get secret %v: %w", secretName, err)
+	}
+	if !ok {
+		return nil
+	}
+	val := string(value)
+	if val == *secret.Value {
```

Comment:
This panics because Value is `nil`.
```suggestion
	if secret.Value != nil && val == *secret.Value {
```

Reply:
forgot to `git push` the latest version 🤦. Now the documentation was also bumped appropriately